src/reader.rs

```diff
@@ -145,6 +146,29 @@ pub enum StreamItem {
     Nothing,
 }
 
+impl StreamItem {
+    /// If `is_null` is `true`, returns `StreamItem::Value(ion_type)`. Otherwise,
+    /// returns `StreamItem::Null(ion_type)`.
+    pub fn nullable_value(ion_type: IonType, is_null: bool) -> StreamItem {
+        if is_null {
+            StreamItem::Null(ion_type)
+        } else {
+            StreamItem::Value(ion_type)
+        }
+    }
+}
+
+impl Display for StreamItem {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        use StreamItem::*;
+        match self {
+            Value(ion_type) => write!(f, "null.{}", ion_type),
+            Null(ion_type) => write!(f, "{}", ion_type),
```

Comment:
Looks like these two got swapped:
```suggestion
            Value(ion_type) => write!(f, "{}", ion_type),
            Null(ion_type) => write!(f, "null.{}", ion_type),
```